app.ts

```diff
@@ -0,0 +1,149 @@
+//#region AppLogic
+//#region API methods
+function getRecordCount() {
+    return fetch('http://localhost:2050/recordCount')
+        .then(function (response) {
+            return response.json();
+        });
+}
+function getColumnNames() {
+    return fetch('http://localhost:2050/columns')
+        .then(function (response) {
+            return response.json();
+        });
+}
+async function getRecords(fromID: number, toID: number) {
+    const request = 'http://localhost:2050/records?from=' + (fromID - 1).toString() + '&to=' + (toID - 1).toString();
+    return await fetch(request)
+        .then(function (response) {
+            return response.json();
+        });
+}
+//#endregion
+
+//#region Data Loading methods
+function InsertValue(value: string, id: string) {
+    const writable: string = '<td id=\"' + id + '\" align="center">' + value.toString() + '</td>';
+    $("#ContentBody").append(writable);
+}
+function UpdateColumnNames(value: string) {
+    const writable: string = '<th align="center">' + value.toString() + '</th>';
+    $("#HeaderRow").append(writable);
+}
+function placeRecords(fromID: number, toID: number) {
+    $("#ContentBody").empty();
+    getRecords(fromID, toID)
+        .then((records: Array<Array<string>>) => {
+            for (var rowIndex in records) {
+                $("#ContentBody").append('<tr id="' + records[rowIndex][0] + '">');
+                for (var columnIndex in records[rowIndex]) {
+                    InsertValue(records[rowIndex][columnIndex], records[rowIndex][0]);
+                }
+                $("#ContentBody").append('</tr>');
+            }
+        });
+    return [fromID as number, toID as number];
+}
+
+function placeRecordsFromCursor(cursor: Array<number>) {
+    cursor = cursor.sort((a,b) => {return a-b});
+    return placeRecords(cursor[0], cursor[1])
+}
+//#endregion
+
+//#region Handlers
+function getPageContent(fromID: number, toID: number) {
+    const cursor = [fromID as number, toID as number];
+    $("#HeaderRow").empty();
+    getColumnNames()
+        .then((columns: Array<string>) => {
+            for (var columnIndex in columns) {
+                UpdateColumnNames(columns[columnIndex]);
+            }
+        });
+    placeRecords(fromID, toID);
+    return cursor;
+}
+
+function toNumber(input: string | number) {
+    if (typeof input === "string") {
+        return parseInt(input);
+    }
+    else if (typeof input === "number") {
```

Comment:
You could actually remove all the `else` and `else if` usage since the `return` takes care of that. This is stylistic though, not sure what @FritzOnFire and @JacoBriers think?

Reply:
Hmmm... my preference here would be a case statement. But yes, because we return, we don't need the else.

Reply:
👍 @JacoBriers. I try not to think about switches in JavaScript unless I have more than 3/4 cases because they fallthrough by default 😢 However, I realise this can be circumvented by placing them in their own function and using returns to make things look nicer. In this specific case a switch is perfect because that isn't even necessary 😄